mailpoet/assets/js/src/segments/dynamic/dynamic-segments-filters/fields/woocommerce/purchased-with-attribute.tsx

```diff
@@ -90,23 +137,52 @@ export function PurchasedWithAttributeFields({
         dimension="small"
         key="select-segment-product-attribute"
         placeholder={__('Search attributes', 'mailpoet')}
-        options={productAttributesOptions}
-        value={filter((productAttributeOption) => {
-          if (segment.attribute_taxonomy_slug === undefined) {
-            return undefined;
-          }
-          return (
-            segment.attribute_taxonomy_slug === productAttributeOption.value
-          );
-        }, productAttributesOptions)}
+        options={combinedOptions}
+        value={
+          segment.attribute_type === 'local'
+            ? filter((localAttributeOption) => {
+                if (!segment.attribute_local_name) {
+                  return undefined;
+                }
+                return (
+                  `${segment.attribute_local_name}@local` ===
+                  localAttributeOption.value
+                );
+              }, localAttributeOptions)
+            : filter((productAttributeOption) => {
+                if (segment.attribute_taxonomy_slug === undefined) {
+                  return undefined;
+                }
+                return (
+                  segment.attribute_taxonomy_slug ===
+                  productAttributeOption.value
+                );
+              }, productAttributesOptions)
+        }
```

Comment:
Maybe it could be slightly more readable to store the value into a variable before the JSX code. I think it would be easier to read (possibly also cache with useMemo) out of the JSX code, and also, the JSX would probably be more readable.

Reply:
I ended up extracting most things and then memoized a lot of stuff to make sure we're avoiding re-renders. Let me know what you think.